library/core/resource-loader/src/main/java/org/quiltmc/qsl/resource/loader/api/ResourcePackActivationType.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.resource.loader.api;
+
+/**
+ * Represents the resource pack activation type.
+ */
+public enum ResourcePackActivationType {
+	/**
+	 * Normal activation. The user has full control over the activation of the resource pack.
+	 */
+	NORMAL,
+	/**
+	 * Enabled by default. The user has still full control over the activation of the resource pack.
+	 *
+	 * <p>Note: this setting can only be satisfied on data packs, client resource packs cannot be by default enabled.
+	 */
+	DEFAULT_ENABLED,
+	/**
+	 * Always enabled. The user cannot disable the resource pack.
+	 */
+	ALWAYS_ENABLED;
+
+	/**
+	 * Returns whether this resource pack will be enabled by default or not.
+	 *
+	 * @return {@code true} if enabled by default, else {@code false}
+	 */
+	public boolean isEnabledByDefault() {
```

Comment:
We've dropped get/is prefixes in Loader. Should we do the same here?
cc @i509VCB

Reply:
Not just a plain drop, sometimes get/is is required depending on the context, but yes.

Reply:
I'd say this is a case where `enabledByDefault` is fine.

Reply:
Guessing this was a no for the response?